packages/firestore/src/api/database.ts

```diff
@@ -552,6 +553,61 @@ export class WriteBatch
   }
 }
 
+class UntypedFirestoreDataConverterAdapter<U>
+  extends Compat<PublicFirestoreDataConverter<U>>
+  implements UntypedFirestoreDataConverter<U> {
+  private static readonly INSTANCES = new WeakMap();
+
+  private constructor(
+    private readonly _firestore: Firestore,
+    delegate: PublicFirestoreDataConverter<U>
+  ) {
+    super(delegate);
+  }
+
+  fromFirestore(snapshot: unknown, options?: unknown): U {
```

Comment:
Can you use `ExpQueryDocumentSnapshot` and `PublicSnapshotOptions` here (instead of unknown)?